File: Python/Practices/financial_calculator.py
# JF Financial Calculator

def g(p): return round(float(input(f"\nWhat is your monthly {p}? $")),2)
def c(m,v,t): print(f"Your {m} is {v/t*100}% of your income.")
i=g("income")
p=i-sum(c(m,v:=g(m),i) or v for m in ["rent", "utilities", "groceries", "transportation"])
print(f"\nYou should save ${i/10} a month, that is 10% of your income.\n\nYou have ${p-i/10} of spending money each month!")

File: C/Practices/financial_calculator.c
// JF, Financial Calculator

#include <stdio.h>

float get_expense(char* prompt){
    float expense;
    printf("What is your monthly %s? $", prompt);
    scanf("%f", &expense);
    return expense;
}

void calc_percent(char* category, float expense, float income){
    printf("Your %s is %.2f percent of your income.\n\n", category, expense/income*100);
}

int main(){
    float income;
    printf("What is your monthly income? $");
    scanf("%f", &income);
    printf("\n");

    float sum;
    char categories[4][15] = {"rent", "utilities", "groceries", "transportation"};

    for (int i=0; i<4; i++) {
        float expense = get_expense(categories[i]);
        calc_percent(categories[i], expense, income);
        sum += expense;
    }

    printf("You should save $%.2f a month, that is 10 percent of your income.\n\nYou have $%.2f of spending money each month!", income/10, 0.9*income - sum);

    return 0;
}